custom/abdm/hiu/serializers/base.py

```diff
@@ -0,0 +1,64 @@
+from rest_framework import serializers
+
+from custom.abdm.const import TIME_UNITS, DATA_ACCESS_MODES, GATEWAY_CONSENT_STATUS_CHOICES, CONSENT_PURPOSES
+
+
+class GatewayRequestBaseSerializer(serializers.Serializer):
+    requestId = serializers.UUIDField()
+    timestamp = serializers.DateTimeField()
+
+
+class GatewayErrorSerializer(serializers.Serializer):
+    code = serializers.IntegerField()
+    message = serializers.CharField()
+
+
+class GatewayResponseReferenceSerializer(serializers.Serializer):
+    requestId = serializers.UUIDField()
+
+
+class GatewayIdSerializer(serializers.Serializer):
+    id = serializers.CharField()
+
+
+class GatewayCareContextSerializer(serializers.Serializer):
+    patientReference = serializers.CharField()
+    careContextReference = serializers.CharField()
+
+
+class GatewayRequesterSerializer(serializers.Serializer):
+    class IdentifierSerializer(serializers.Serializer):
+        type = serializers.CharField()
+        value = serializers.CharField()
+        system = serializers.CharField(required=False, allow_null=True)
+
+    name = serializers.CharField()
+    identifier = IdentifierSerializer(required=False)
+
+
+class GatewayPermissionSerializer(serializers.Serializer):
+    class DateRangeSerializer(serializers.Serializer):
+        vars()['from'] = serializers.DateTimeField()
```

Comment:
I am using DRF Serializers to validate request body. ABDM request body is nested with upto 3 layers and hence the above nested serializers . 
 `        vars()['from'] = serializers.DateTimeField()
`
In case you are asking about, `from` is a reserved keyword in django and hence this to get field name from local scope.

Reply:
yup, was wondering what the form business was.